hack/set-motd/main_test.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"testing"
+)
+
+func Test_setMotd(t *testing.T) {
+
+	// Create (or clean) testfile
+	testFile := "testfile"
+	f, err := os.Create(testFile)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	type args struct {
+		motd       *stringFlag
+		user       *stringFlag
+		pr         *stringFlag
+		customPath *stringFlag
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    [5]string
+		wantErr bool
+	}{
+		// Tests array
+		{
+			"Test 1: Contains only date. No error returned expected.",
+			args{
+				motd:       &stringFlag{false, ""},
+				user:       &stringFlag{false, ""},
+				pr:         &stringFlag{false, ""},
+				customPath: &stringFlag{true, f.Name()},
+			},
+			[5]string{"Updated at", "", "", "", ""},
+			false,
+		},
+		{
+			"Test 2: Contains date and motd. No error returned expected.",
+			args{
+				motd:       &stringFlag{true, "OCATOPIC"},
+				user:       &stringFlag{false, ""},
+				pr:         &stringFlag{false, ""},
+				customPath: &stringFlag{true, f.Name()},
+			},
+			[5]string{"Updated at", "OCATOPIC", "", "", ""},
+			false,
+		},
+		{
+			"Test 2: Contains nothing. Error returned expected.",
+			args{
+				motd:       &stringFlag{false, "false"},
+				user:       &stringFlag{false, "false"},
+				pr:         &stringFlag{false, "false"},
+				customPath: &stringFlag{false, f.Name()},
+			},
+			[5]string{"false", "false", "false", "false", "false"},
+			true,
+		},
+	}
+
+	for i, currentTest := range tests {
+		t.Run(currentTest.name, func(t *testing.T) {
+			setMotd(currentTest.args.motd, currentTest.args.user, currentTest.args.pr, currentTest.args.customPath)
```

Comment:
As a note, instead of a path, you could expect a [`Writer`](https://pkg.go.dev/io#Writer) interface. This would ease the testing since you could just implement your own test `Writer` interface or use [`bytes.Buffer`](https://pkg.go.dev/bytes#Buffer), avoiding working with files.